include/mbedtls/config_psa.h

```diff
@@ -298,94 +298,94 @@ extern "C" {
 #endif /* PSA_WANT_ALG_TLS12_ECJPAKE_TO_PMS */
 
 #if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_USE)
+#define PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY 1
 #if !defined(MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_USE)
 #define MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR_USE 1
-#define PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY 1
 #endif /* !MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_USE */
 #endif /* PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_USE */
 
 #if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT)
+#define PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY 1
```

Comment:
I this really necessary? `IMPORT` (or any other symbol) implies `USE`, so we should only need this line in the `USE` block.

I think we should have the non-USE blocks first, each enabling the `USE` macro, then after those the `USE` block, in turn enabling `PUBLIC_KEY`.

Reply:
> IMPORT (or any other symbol) implies USE

I don't remember me adding this kind of check in the code. Was it expected? It makes sense indeed, but I didn't enforced that

Reply:
The idea is that any creation method implies you can use the key. That wasn't part of this pull request's objective, but we can take advantage of it.

Reply:
I think it was expected: from [the comment linked to in the description](https://github.com/Mbed-TLS/mbedtls/issues/7439#issuecomment-1543790495): 

> The documented "auto-enable graph" would be very simple: ALL enables IMPORT, EXPORT, GENERATE, DERIVE, and any of those 4 enable USE.

In the meantime, we decided against including an `ALL` symbol after all, but I think the part where any of IMPORT, EXPORT, GENERATE or DERIVE auto-enables USE is still valid. I completely forgot to reflect that in the [issue description](https://github.com/Mbed-TLS/mbedtls/issues/7614) though, sorry for that.

Also, now that I think of it, do we really need the internal `LEGACY` symbols then? Can't we just use the new `USE` symbol instead? I mean, since you fixed the logic flaw pointed out by Gilles (thanks for noticing btw!), `LEGACY` is defined if any of USE, IMPORT, EXPORT, GENERATE or DERIVE is defined, which, once the auto-enable behaviour described above is implemented, will be equivalent to USE being defined. So `LEGACY` is equivalent to `USE` and we don't need `LEGACY` at all. @gilles-peskine-arm @valeriosetti Am I missing something? In particular, if my reasoning is correct for `PSA_WANT` macros, is it also correct for `MBEDTLS_PSA_BUILTIN` macros?

Reply:
> do we really need the internal LEGACY symbols then? Can't we just use the new USE symbol instead?

I don't see how that would work. LEGACY will be replaced by different things in different places: USE when doing an operation, EXPORT when exporting, etc.